Source/SvgDocument.cs

```diff
@@ -175,6 +175,68 @@ public virtual SvgElement GetElementById(string id)
             return (this.GetElementById(id) as TSvgElement);
         }
 
+        /// <summary>
+        /// Validate whether the system has GDI+ capabilities (non Windows related).
+        /// </summary>
+        /// <returns>Boolean whether the system is capable of using GDI+</returns>
+        public static bool SystemIsGdiPlusCapable()
+        {
+            try 
+            {
+                EnsureSystemIsGdiPlusCapable();
+            }
+            catch(SvgGdiPlusCannotBeLoadedException)
+            {
+                return false;
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Ensure that the running system is GDI capable, if not this will yield a
+        /// SvgGdiPlusCannotBeLoadedException exception.
+        /// </summary>
+        public static void EnsureSystemIsGdiPlusCapable()
+        {
+            try
+            {
+                var matrix = new Matrix(0, 0, 0, 0, 0, 0);
+            }
+            // GDI+ loading errors will result in TypeInitializationExceptions, 
+            // for readability we will catch and wrap the error
+            catch (Exception e)
+            {
+                if (ExceptionCaughtIsGdiPlusRelated(e))
+                {
+                    // Throw only the customized exception if we are sure GDI+ is causing the problem
+                    throw new SvgGdiPlusCannotBeLoadedException(e);
+                }   
+                //Ignore any other form of error
+            }
+        }
+
+        /// <summary>
+        /// Check if the current exception or one of its children is the targeted GDI+ exception. 
+        /// It can be hidden in one of the InnerExceptions, so we need to iterate over them.
+        /// </summary>
+        /// <param name="e">The exception to validate against the GDI+ check</param>
+        private static bool ExceptionCaughtIsGdiPlusRelated(Exception e)
+        {
+            var currE = e;
+            int cnt = 0; // Keep track of depth to prevent endless-loops
+            while (currE != null && cnt < 10)

```

Comment:
Have you exceeded 10 in your experience ?

Reply:
Unfortunately yes (not in this library), since we get an error from a deeper library, we cannot always be sure that the content of the exception is "safe". There is a slight possiblity that somewhere somebody wired 2 exceptions as innerexceptions of each other causing an loop, also if we encounter issues like stack of memory issues, there is a possibility that the exception has a lot of inner exceptions, so I guess it's safe to stop at 10. I won't expect it to be deeper than 4 or 5, but if we have to look beyond 10 children, I think we can assume we have a bigger issues (and most likely not a GDI+ issue).